pkg/middleware/cookie_refresh.go

```diff
@@ -0,0 +1,60 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/justinas/alice"
+	middlewareapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/middleware"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests"
+)
+
+type CookieRefreshOptions struct {
+	IssuerURL string
+}
+
+func NewCookieRefresh(opts *CookieRefreshOptions) alice.Constructor {
+	cr := &cookieRefresh{
+		HttpClient: &http.Client{},
+		IssuerURL:  opts.IssuerURL,
+	}
+	return cr.refreshCookie
+}
+
+type cookieRefresh struct {
+	HttpClient *http.Client
+	IssuerURL  string
+}
+
+func (cr *cookieRefresh) refreshCookie(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		scope := middlewareapi.GetRequestScope(req)
+		if scope.Session == nil || !scope.Session.SessionJustRefreshed {
+			next.ServeHTTP(rw, req)
+			return
+		}
+
+		cookie, err := req.Cookie("hsdpamcookie")
```

Comment:
Could the cookie name be configurable? I know the IAM cookie is named `hsdpamcookie` but it would be interesting because other people could easily use it even if they are not authenticating via IAM and also if eventually, IAM changes the cookie name we don't need a new version of the proxy to fix it.